src/attachments/attachments.js

```diff
@@ -0,0 +1,76 @@
+const request = require("request-promise-native");
+
+const randomId = require("../utils/randomId");
+
+const addAttachment = async (
+  apiClient,
+  { taskId, commentText, attachmentUrl, attachmentFileName }
+) => {
+  const comment = {
+    id: randomId(),
+    task_id: taskId,
+    body: commentText,
+  };
+
+  const attachment = {
+    id: randomId(),
+    parent_id: comment.id,
+  };
+
+  const attachmentVersion = {
+    id: randomId(),
+    attachment_id: attachment.id,
+    name: attachmentFileName,
+  };
+
+  try {
+    /* The `request` library is deprecated, however it was the method I was able
+      to use to fetch the file from the URL and send it to the /files endpoint.
+      It may need a refactoring at some point.
+    */
+    const file = await request.get(attachmentUrl);
+
+    await apiClient.post("sync", {
+      comments: {
+        created: [{ ...comment }],
+        updated: [],
+        deleted: [],
+      },
+      attachments: {
+        created: [{ ...attachment }],
+        updated: [],
+        deleted: [],
+      },
+      attachment_versions: {
+        created: [{ ...attachmentVersion }],
+        updated: [],
+        deleted: [],
+      },
+    });
+
+    let formData = {};
+    formData[attachmentVersion.id] = {
+      value: file,
+      options: {
+        filename: attachmentVersion.name,
+      },
+    };
+
+    const { baseURL, headers } = apiClient.defaults;
+
+    await request.post({
```

Comment:
I don't fully understand why not use `global.fetch()` (+ a polyfill if needed), but alright.

Reply:
Well, `fetch` is not available in Node.js itself, and I thought we want to support also the Node projects (like Electron apps?). IMO we could ditch both `request` (as it is obsolete) and `axios` in favor of `node-fetch`. Thoughts?

Reply:
I don't have much experience with Node/Electron stuff, but `node-fetch` sounds like what I meant by saying `+ polyfill if needed` :) I don't really think that it's necessarily worth your time right now to refactor this, but in the long run, yes, that sounds like the way to go